utilities/test_suite/HOST/runTests.py

```diff
@@ -154,15 +236,135 @@ def rpp_test_suite_parser_and_validator():
     numRuns = 1
 elif(testType == 1):
     if numRuns == 0:
-        numRuns = 100  #default numRuns for running performance tests
+        numRuns = 100 #default numRuns for running performance tests
     outFilePath = os.path.join(os.path.dirname(cwd), 'OUTPUT_PERFORMANCE_LOGS_HOST_' + timestamp)
+else:
+    print("Invalid TEST_TYPE specified. TEST_TYPE should be 0/1 (0 = Unittests / 1 = Performancetests)")
+    exit()
+
+os.mkdir(outFilePath)
+loggingFolder = outFilePath
 dstPath = outFilePath
 
-# run the shell script
-subprocess.call(["./testAllScript.sh", srcPath1, args.input_path2, str(testType), str(numRuns), str(qaMode), str(decoderType), str(preserveOutput), str(batchSize), " ".join(caseList)])  # nosec
+# Validate DST_FOLDER
+validate_and_remove_contents(dstPath)
+
+# Enable extglob
+if os.path.exists("build"):
+    shutil.rmtree("build")
+os.makedirs("build")
+os.chdir("build")
+
+# Run cmake and make commands
+subprocess.run(["cmake", ".."])
+subprocess.run(["make", "-j16"])
+
+print("\n\n\n\n\n")
+print("##########################################################################################")
+print("Running all layout Inputs...")
+print("##########################################################################################")
+
+if testType == 0:
+    for case in caseList:
+        if int(case) < 0 or int(case) > 86:
+            print(f"Invalid case number {case}. Case number must be in the range of 0 to 86!")
+            continue
+        for layout in range(3):
+            dstPathTemp, log_file_layout = process_layout(layout, qaMode, case, dstPath)
+
+            if qaMode == 0:
+                if not os.path.isdir(dstPathTemp):
+                    os.mkdir(dstPathTemp)
+
+            print("\n\n\n\n")
+            print("--------------------------------")
+            print("Running a New Functionality...")
+            print("--------------------------------")
+
+            for bitDepth in range(7):
+                print("\n\n\nRunning New Bit Depth...\n-------------------------\n\n")
+
+                for outputFormatToggle in range(2):
+                    # There is no layout toggle for PLN1 case, so skip this case
+                    if layout == 2 and outputFormatToggle == 1:
+                        continue
+
+                    if case == "8":
```

Comment:
Please add more comments as much as possible:
```
# Run all variants of noise type functions with additional argument of noiseType = gausssianNoise / shotNoise / poisssonNoise
```

Reply:
Sorry may have missed salt/pepper noise. Please check types and add

Reply:
Done